src/backend/distributed/transaction/multi_shard_transaction.c

```diff
@@ -95,11 +85,11 @@ OpenTransactionsToAllShardPlacements(List *shardIntervalList, char *userName)
 									   shardPlacement->nodePort)));
 			}
 
-			connection = StartNodeUserDatabaseConnection(FORCE_NEW_CONNECTION,
-														 shardPlacement->nodeName,
-														 shardPlacement->nodePort,
-														 userName,
-														 NULL);
+			connection = StartPlacementConnection(FOR_DDL,
```

Comment:
Hm. Is FOR_DDL unconditionally right here? I think it might be better to pass that in, so we can use it both for INSERT ... SELECT and DDL?